features/Proxy.feature

```diff
@@ -4,17 +4,17 @@ Feature: Proxy requests to upstream.
+        Given request JSON payload '{"name":"posts","active":true,"proxy":{"preserve_host":false,"listen_path":"/posts/*","upstreams": {"balancing": "roundrobin", "targets": [{"target": "http://localhost:9089/posts"}]},"strip_path":true,"append_path":false,"enable_load_balancing":false,"methods":["ALL"],"hosts":["hellofresh.*"]},"plugins":[{"name":"cors","enabled":true,"config":{"domains":["*"],"methods":["GET","POST","PUT","PATCH","DELETE"],"request_headers":["Origin","Authorization","Content-Type"],"exposed_headers":["X-Debug-Token","X-Debug-Token-Link"]}},{"name":"rate_limit","enabled":true,"config":{"limit":"10-S","policy":"local"}},{"name":"oauth2","enabled":true,"config":{"server_name":"local"}},{"name":"compression","enabled":true}]}'
         When I request "/apis" API path with "POST" method
         Then I should receive 201 response code
         And header "Location" should be "/apis/posts"
 
-        Given request JSON payload '{"name":"posts-public","active":true,"proxy":{"preserve_host":false,"listen_path":"/posts-public/*","upstream_url":"http://localhost:9089/posts","strip_path":true,"append_path":false,"enable_load_balancing":false,"methods":["ALL"]},"plugins":[{"name":"cors","enabled":true,"config":{"domains":["*"],"methods":["GET","POST","PUT","PATCH","DELETE"],"request_headers":["Origin","Authorization","Content-Type"],"exposed_headers":["X-Debug-Token","X-Debug-Token-Link"]}},{"name":"rate_limit","enabled":true,"config":{"limit":"10-S","policy":"local"}},{"name":"oauth2","enabled":true,"config":{"server_name":"local"}},{"name":"compression","enabled":true}]}'
+        Given request JSON payload '{"name":"posts-public","active":true,"proxy":{"preserve_host":false,"listen_path":"/posts-public/*","upstreams": {"balancing": "roundrobin", "targets": [{"target": "http://localhost:9089/posts"}]},"strip_path":true,"append_path":false,"enable_load_balancing":false,"methods":["ALL"]},"plugins":[{"name":"cors","enabled":true,"config":{"domains":["*"],"methods":["GET","POST","PUT","PATCH","DELETE"],"request_headers":["Origin","Authorization","Content-Type"],"exposed_headers":["X-Debug-Token","X-Debug-Token-Link"]}},{"name":"rate_limit","enabled":true,"config":{"limit":"10-S","policy":"local"}},{"name":"oauth2","enabled":true,"config":{"server_name":"local"}},{"name":"compression","enabled":true}]}'
```

Comment:
If possible it would be better to have smth like:
```
Given request JSON payload '{
  "name":"posts-public",
  "active":true,
  "proxy":{
    "preserve_host":false,
    "listen_path":"/posts-public/*",
    "upstreams":{
      "balancing":"roundrobin",
      "targets":[
        {
          "target":"http://localhost:9089/posts"
        }
      ]
    },
    "strip_path":true,
    "append_path":false,
    "enable_load_balancing":false,
    "methods":[
      "ALL"
    ]
  },
  "plugins":[
    {
      "name":"cors",
      "enabled":true,
      "config":{
        "domains":[
          "*"
        ],
        "methods":[
          "GET",
          "POST",
          "PUT",
          "PATCH",
          "DELETE"
        ],
        "request_headers":[
          "Origin",
          "Authorization",
          "Content-Type"
        ],
        "exposed_headers":[
          "X-Debug-Token",
          "X-Debug-Token-Link"
        ]
      }
    },
    {
      "name":"rate_limit",
      "enabled":true,
      "config":{
        "limit":"10-S",
        "policy":"local"
      }
    },
    {
      "name":"oauth2",
      "enabled":true,
      "config":{
        "server_name":"local"
      }
    },
    {
      "name":"compression",
      "enabled":true
    }
  ]
}'
When I request "/apis" API path with "POST" method
Then I should receive 201 response code
And header "Location" should be "/apis/posts-public"
```

I used https://jsonformatter.curiousconcept.com

Reply:
This is a question for @vgarvardt. But honestly it's not a problem at the moment

Reply:
It is possible to use multiline parameters, but requires some handlers modifications. I can add this later and change all the steps with json.

Reply:
Np thx Vlad 👍